src/shims/unix/android/foreign_items.rs

```diff
@@ -25,8 +26,65 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_scalar(errno_place.to_ref(this).to_scalar(), dest)?;
             }
 
+            // Threading
+            "prctl" => {
+                // We do not use `check_shim` here because `prctl` is variadic. The argument
+                // count is checked bellow.
+                this.check_abi_and_shim_symbol_clash(abi, Abi::C { unwind: false }, link_name)?;
+
+                check_args_len("prctl", args, 1)?;
+
+                let id = this.read_scalar(&args[0])?.to_i32()?;
+                // FIXME: Use PR_SET_NAME and PR_GET_NAME constants when
+                // https://github.com/rust-lang/libc/pull/3941 lands.
+                const PR_SET_NAME: i32 = 15;
+                const PR_GET_NAME: i32 = 16;
+
+                let res = match id {
+                    PR_SET_NAME => {
+                        check_args_len("'PR_SET_NAME' prctl", args, 2)?;
+
+                        let tid = this.pthread_self()?;
+                        let name = this.read_scalar(&args[1])?;
+                        let name_len = 16;
+
+                        this.pthread_setname_np(tid, name, name_len)?
+                    }
+                    PR_GET_NAME => {
+                        check_args_len("'PR_GET_NAME' prctl", args, 2)?;
+
+                        let tid = this.pthread_self()?;
+                        let name = this.read_scalar(&args[1])?;
+                        let name_len = Scalar::from_target_usize(16, this);
+
+                        this.pthread_getname_np(tid, name, name_len)?
+                    }
+                    _ => {
+                        this.handle_unsupported_foreign_item(format!(
+                            "can't execute prctl with ID {id}"
+                        ))?;
+                        return interp_ok(EmulateItemResult::AlreadyJumped);
+                    }
+                };
+                this.write_scalar(res, dest)?;
+            }
+
             _ => return interp_ok(EmulateItemResult::NotSupported),
         }
         interp_ok(EmulateItemResult::NeedsReturn)
     }
 }
+
+fn check_args_len<'tcx>(
```

Comment:
We already have `check_arg_count` in helpers.rs. So maybe name this `check_min_arg_count` and also put it in that file?

Seems like it would also make sense to make this return an array of the right size, like `check_arg_count`.